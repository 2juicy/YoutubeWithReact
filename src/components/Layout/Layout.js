import React, { useEffect } from "react";
import "./Layout.scss";
import Header from "../../containers/Header/Header";
import { useLocation } from "react-router-dom";
import { usePrevious } from "../../hooks/usePrevious";

export default function Layout(props) {
  const location = useLocation();
  const prevLocation = usePrevious(location);

  useEffect(() => {
    if (location !== prevLocation && window) window.scrollTo(0, 0);
  });

  return (
    <div className="layout">
      <Header />
      {props.children}
    </div>
  );
}
