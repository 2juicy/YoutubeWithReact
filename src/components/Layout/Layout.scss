@import "../../styles/shared.scss";

.layout {
  & > *:not(:first-child) {
    margin-top: $header-height;
  }

  ::-webkit-scrollbar {
    width: 6px;
  }
}
