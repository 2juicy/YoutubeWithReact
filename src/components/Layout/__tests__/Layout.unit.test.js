import React from "react";
import { shallow } from "enzyme";
import Layout from "../Layout";

describe("Layout", () => {
  test("renders without props", () => {
    const wrapper = shallow(<Layout />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with one child", () => {
    const wrapper = shallow(
      <Layout>
        <div>Child</div>
      </Layout>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with children", () => {
    const wrapper = shallow(
      <Layout>
        <div>Child</div>
        <div>
          <h1>Child</h1>
          <p>Child</p>
        </div>
      </Layout>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
