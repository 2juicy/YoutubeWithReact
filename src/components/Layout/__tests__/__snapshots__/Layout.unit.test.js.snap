// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders with children 1`] = `
<div
  className="layout"
>
  <Header />
  <div>
    Child
  </div>
  <div>
    <h1>
      Child
    </h1>
    <p>
      Child
    </p>
  </div>
</div>
`;

exports[`Layout renders with one child 1`] = `
<div
  className="layout"
>
  <Header />
  <div>
    Child
  </div>
</div>
`;

exports[`Layout renders without props 1`] = `
<div
  className="layout"
>
  <Header />
</div>
`;
